### DISCUSSION TITLE
Cannot connect Jetson nano 4GB with Create 3

### DISCUSSION THREAD

**Discussion by mynameissagif** [iRobotEducation/create3_docs#399] [Questions] [answered]
### How are you connecting to your Create 3?

Wi-Fi and USB Ethernet (ROS 2)

### Computer(s) Model(s) and Operating System(s)

Jetson nano 4GB

### Which version of ROS 2 is installed on your computer?

Galactic

### Which firmware version is installed on your robot?

G.4.5

### Which RMW is your robot running?

CycloneDDS

### Are there multiple Create 3 robots connected to your network?

No

### Is multicast enabled?

No

### What is the Adapter Board's USB/BLE Toggle currently switched to?

USB

### Describe your question.

Hi, I have read other discussions and found similar problems, but none of the suggested solutions solved my problem.
I'm trying to connect my Jetson nano 4GB to the create 3. I've followed the [official docs](https://iroboteducation.github.io/create3_docs/setup/jetson/) to setup the Jetson nano but eventually I can only see /parameter_events and /rosout when typing ros2 topic list. 

here is a total summary of my current state:
when running`scripts/docker_run.sh -c dustynv/ros:galactic-desktop-l4t-r35.3.1`, I get 
`localuser:root being added to access control list
xauth:  file /tmp/.docker.xauth does not exist
sourcing   /opt/ros/galactic/install/setup.bash
notfound   /ros_deep_learning/install/setup.bash
ROS_DISTRO galactic
ROS_ROOT   /opt/ros/galactic
root@jetson-desktop:/#`

then usig either typeC-Ethernet ([like this](https://github.com/iRobotEducation/create3_docs/discussions/133)) or typeC-microUSB ([like this](https://github.com/iRobotEducation/create3_docs/discussions/133)) both `ifconfig l4tbr0` and `ping 192.168.186.2` work fine, but then inside the container using either fastDDS or CycloneDDS `ros2 topic list` returns only `/parameter_events` and `/rosout`.

**Comment by kwujciak** [comment]
Have you tried setting your RMW to FastDDS?

**Reply by mynameissagif** [reply]
yes. exact same issue (can only see `/parameter_events` and  `/rosout` when typing `ros2 topic list`).

**Reply by kwujciak** [reply]
It shouldn't matter since you only have one robot connected, but maybe try using a domain ID. It could help navigate through some potential traffic. Just make sure they're the same on your Jetson and robot.

**Reply by kwujciak** [reply]
Are you using ROS through Docker or did you do a native install?

**Reply by mynameissagif** [reply]
I'm Using ROS through the Docker following [the official instructions](https://iroboteducation.github.io/create3_docs/setup/jetson/). 

here is a total summary of my current state:
when running`scripts/docker_run.sh -c dustynv/ros:galactic-desktop-l4t-r35.3.1`, I get 
`localuser:root being added to access control list
xauth:  file /tmp/.docker.xauth does not exist
sourcing   /opt/ros/galactic/install/setup.bash
notfound   /ros_deep_learning/install/setup.bash
ROS_DISTRO galactic
ROS_ROOT   /opt/ros/galactic
root@jetson-desktop:/#`

then usig either typeC-Ethernet ([like this](https://github.com/iRobotEducation/create3_docs/discussions/133)) or typeC-microUSB ([like this](https://github.com/iRobotEducation/create3_docs/discussions/133)) both `ifconfig l4tbr0` and `ping 192.168.186.2` work fine, but then inside the container using either fastDDS or CycloneDDS with domain ID 32 `ros2 topic list` returns only `/parameter_events` and `/rosout`.

**Reply by kwujciak** [reply]
I was very unsuccessful when trying to use ROS through Docker on my Jetson nano. Docker was causing a lot of issues for me that I couldn't quite work around. What ended up consistently working for me was natively installing ROS 2. I flashed [this](https://github.com/Qengineering/Jetson-Nano-Ubuntu-20-image) image onto a 32 GB mini SD card. Then I used [these iRobot instructions](https://iroboteducation.github.io/create3_docs/setup/ubuntu2004/) to download ROS 2 (Galactic) onto the board. I had both my Jetson and robot on Cyclone (when it was on Fast, we were able to ping the robot but couldn't get a topic list - what you're describing), then followed the [ethernet over usb](https://iroboteducation.github.io/create3_docs/setup/jetson/#d-set-up-ethernet-over-usb) instructions (I did not need a domain ID). Another process that worked was [setting up over wifi](https://iroboteducation.github.io/create3_docs/setup/jetson/#c-setup-wi-fi-connections), but I needed to switch to FastDDS and disable multicast for that to successfully connect. Both ways yielded a full topic list.

**Reply by kwujciak** [reply, marked as answer]
In doing so more testing, it seems that you may have to reboot the robot. So if you're able to ping `192.168.186.2`, navigate to that IP in your Jetson browser, then reboot the robot under application settings. After doing this in the Docker, we were able to get a full topic list.

**Reply by mynameissagif** [reply]
thanks! Using CycloneDDS & typeC-microUSB and rebooting the robot from the Jetson after connecting the cable I've got the proper `ros2 topic list` response. I'll consider moving to the 20.04 Ubuntu image if more issues would rise.
